modules/memberships/class-jetpack-memberships.php

```diff
@@ -0,0 +1,289 @@
+<?php
+/**
+ * Jetpack_Memberships: wrapper for memberships functions.
+ *
+ * @package    Jetpack
+ * @since      7.3.0
+ */
+
+/**
+ * Class Jetpack_Memberships
+ * This class represents the Memberships functionality.
+ */
+class Jetpack_Memberships {
+	/**
+	 * CSS class prefix to use in the styling.
+	 *
+	 * @var string
+	 */
+	public static $css_classname_prefix = 'jetpack-memberships';
+	/**
+	 * Our CPT type for the product (plan).
+	 *
+	 * @var string
+	 */
+	public static $post_type_plan = 'jp_mem_plan';
+	/**
+	 * Option that will store currently set up account (Stripe etc) id for memberships.
+	 *
+	 * @var string
+	 */
+	public static $connected_account_id_option_name = 'jetpack-memberships-connected-account-id';
+	/**
+	 * Button block type to use.
+	 *
+	 * @var string
+	 */
+	private static $button_block_name = 'membership-button';
+	/**
+	 * Classic singleton pattern
+	 *
+	 * @var Jetpack_Memberships
+	 */
+	private static $instance;
+
+	/**
+	 * Jetpack_Memberships constructor.
+	 */
+	private function __construct() {}
+
+	/**
+	 * The actual constructor initializing the object.
+	 *
+	 * @return Jetpack_Memberships
+	 */
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+			self::$instance->register_init_hook();
+		}
+
+		return self::$instance;
+	}
+	/**
+	 * Get the map that defines the shape of CPT post. keys are names of fields and
+	 * 'meta' is the name of actual WP post meta field that corresponds.
+	 *
+	 * @return array
+	 */
+	private static function get_plan_property_mapping() {
+		$meta_prefix = 'jetpack_memberships_';
+		$properties  = array(
+			'price'    => array(
+				'meta' => $meta_prefix . 'price',
+			),
+			'currency' => array(
+				'meta' => $meta_prefix . 'currency',
+			),
+		);
+		return $properties;
+	}
+
+	/**
+	 * Transform WP CPT post into array representing a memberships product.
+	 *
+	 * @param WP_Post $product_post - CPT representing the product.
+	 * @return array
+	 */
+	public static function product_post_to_array( $product_post ) {
```

Comment:
This file is verbatim in wpcom codebase and it's used there.

Reply:
I seem to see 2 copies on WordPress.com, one using that function and the other that does not. Is that on purpose?

Reply:
Yes. One is old (mu-plugins/memberships) and with multiple PRs I am removing all usages so I can finally delete it :)

Reply:
Alright. In this case I guess we'll need to either update the newest copy so it includes the shortcode (that uses the `product_post_to_array` function above) and update this PR accordingly, or remove `product_post_to_array` since it does not appear to be in use in the newer version?

Reply:
You are right, we should be able to safely remove this entirely. Will do!

Reply:
Function removed. https://github.com/Automattic/jetpack/pull/9802/commits/e018d63b46e84506af64ce0590373fd40745a670